### PR TITLE
Course creation PR to make a Code review

### PR DESCRIPTION
**Pull request by cmltaWt0**
@auraz @cjlee112

PR to add Course creation functionality.

### CODE REVIEW

File: mysite/lti/forms.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from django import forms
+
+from lti.models import CourseRef
+
+
+CHOICES = ((0, 'new'), (1, 'saved'))
+
+
+class ChoiceCourseForm(forms.Form):
+    """Form for choising source of new Course"""
```
Comment (auraz) [inline mysite/lti/forms.py]: Typo: choising, rephrase  plz

File: mysite/lti/views.py [hunk 2]
```diff
@@ -119,3 +135,95 @@ def lti_redirect(request, course_id=None, unit_id=None):
...
+    merge content from existing one
+    """
+    lti_post = request.session.get('LTI_POST')
+    course_ref = CourseRef.objects.filter(
+        context_id=pickle.loads(lti_post).get('context_id')
+    ).first()
+    if course_ref:
+        return redirect(reverse('ct:home'))
+
+    parent_courseref = None
```
Comment (auraz) [inline mysite/lti/views.py]: Why we need it here?

File: mysite/lti/views.py [hunk 3]
```diff
@@ -119,3 +135,95 @@ def lti_redirect(request, course_id=None, unit_id=None):
...
+
+def clone_course(user, course):
+    """Clone Course with all related entries
+
+    param: course Course entry
+    return: cloned Course entry
+    """
+    cloned_course = Course(
+        title=course.title, addedBy=user,
+        description=course.description, access=course.access,
```
Comment (auraz) [inline mysite/lti/views.py]: code style plz

File: mysite/lti/views.py [hunk 4]
```diff
@@ -119,3 +135,95 @@ def lti_redirect(request, course_id=None, unit_id=None):
...
+    request_dict = pickle.loads(request.session['LTI_POST'])
+    if not request.session.get('is_valid'):
+        return redirect(reverse('ct:home'))
+    context_id = request_dict.get('context_id')
+    role = ROLES_MAP.get(request_dict.get('roles', None), 'student')
+    # Make sure this context_id is not used
+    course_ref = CourseRef.objects.filter(context_id=context_id).first()
+    if course_ref:
+        if role == 'prof':
+            return redirect(reverse('ct:edit_course', args=(course_ref.course.id,)))
```
Comment (auraz) [inline mysite/lti/views.py:212]: What if I go to this URL manually, as student, will I have prof rights?

File: mysite/lti/views.py (hunk 2 above)
Comment (cmltaWt0) [inline mysite/lti/views.py]: `parent_courseref` needed for creation link to previous revisions of new Course

File: mysite/lti/views.py (hunk 4 above)
Comment (cmltaWt0) [inline mysite/lti/views.py:212]: > What if I go to this URL manually, as student, will I have prof rights?

No, student will be redirected to home page.

File: mysite/lti/forms.py (hunk 1 above)
Comment (cmltaWt0) [inline mysite/lti/forms.py]: > Typo: choising, rephrase plz

Done.

File: mysite/lti/views.py (hunk 2 above)
Comment (auraz) [inline mysite/lti/views.py]: I do not see where it is used below

Comment (cmltaWt0) [inline mysite/lti/views.py]: In line 159 and 162

File: mysite/lti/views.py (hunk 3 above)
Comment (cmltaWt0) [inline mysite/lti/views.py]: > code style plz

Done

File: mysite/lti/views.py (hunk 2 above)
Comment (cmltaWt0) [inline mysite/lti/views.py]: > Why we need it here?

Done

**Pull request closed without merging**